tests/controls/maskededittest.cpp

```diff
@@ -0,0 +1,149 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        tests/controls/maskededittest.cpp
+// Purpose:     wxMaskedEdit controls unit test
+// Author:      Manuel Martin
+// Created:     2012-07-30
+// RCS-ID:      $Id:
+// Copyright:   (c)
+///////////////////////////////////////////////////////////////////////////////
+
+#include "testprec.h"
+
+#if wxUSE_MASKED_EDIT
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/app.h"
+#endif // WX_PRECOMP
+
+#include "wx/maskededit.h"
+
+#include "testableframe.h"
+#include "wx/uiaction.h"
+
+class MaskedEditTestCase : public CppUnit::TestCase
+{
+public:
+    MaskedEditTestCase() { }
+
+    virtual void setUp() wxOVERRIDE
+    {
+        CreateControl();
+    }
+
+    virtual void tearDown() wxOVERRIDE
+    {
+        wxTheApp->GetTopWindow()->DestroyChildren();
+    }
+
+private:
+    CPPUNIT_TEST_SUITE( MaskedEditTestCase );
+    CPPUNIT_TEST( TestSet );
+    WXUISIM_TEST( TestUI );
+    CPPUNIT_TEST( TestFunc );
+    CPPUNIT_TEST_SUITE_END();
+
+    void TestSet();
+#if wxUSE_UIACTIONSIMULATOR
+    void TestUI();
+#endif
+    void TestFunc();
+    void CreateControl();
+    wxMaskedEditText* m_editTx;
```

Comment:
Minor, but "Tx" naming convention is not used anywhere else in wx, so I'd rather not use it here neither. It could be called just `m_edit` or `m_maskedEdit` or `m_maskedEditText`.